README.md

```diff
@@ -0,0 +1,195 @@
+  }
+}
+```
+
+## IntelliJ
+   
+- To run java code you first need to compile it. To compile code in the terminal use command: `javac fileName.java`
+- If the file is successfully compiled it will create a fileName.class file that can be run
+- To run this file use the command: `java fileName`
+
+- You can also run the app with IntelliJ configurations. To build the gradle project make sure you right click on build.gradle and click 'import Gradle Project' after this you can then click the play button on the upper right to run the project
+
+## Comments
+
+- When comments are short we use the single-line syntax: `//`
+- When comments are long we use the multi-line syntax: `/* */`
+
+## Tips
+
+- The controller should be clean of functionality and only contain the endpoints
+- Keep private methods small as they can not be tested directly
+- Keep methods LOOSELY coupled so they are not dependent on each other and are easier to test and scale
+
+## Spring Boot
+
+- Spring is a Java Framework that makes creating Java applications faster and easier:
+
+  - Sets up the boilerplate for an application that gets it up and running quickly
+  - Automates configuration
+  - Spring initializer is used to generate and application with customizable dependencies of your choice
+
+- use Spring Initializr to Bootstrap your application at https://start.spring.io/
+
+- Maven VS Gradle:
+
+  - Maven is a rigid model while Gradle is more flexible and easily customizable
+  - Google is using Gradle as the build system for Android Studio
+  - For more details about these two build automations visit https://dzone.com/articles/gradle-vs-maven
+
+- .war vs .jar:
+  - A .war file is a Web Application Archive which runs inside an application server while a .jar is Java Application Archive that runs a desktop application on a user's machine. A war file is a special jar file that is used to package a web application to make it easy to deploy it on an application server.
+
+### Gradle
+
+- To run a build initially use command `gradle build` to add dependencies
+- Use command `gradle bootRun` to execute file
+- Add the dependency of `implementation 'org.springframework.boot:spring-boot-starter-web'` to have web development dependency this will make your application run on localhost: 8080 when application is executed
+
+- when opening gradle project in IntelliJ two finger click the build.gradle file and click `import gradle project` to import as a gradle project
+
+- Run gradle application using profiles
+
+```sh
+
+gradle bootRun --args="--spring.profiles.active=local" # local is the profile name in this example which is application-local.yml
+
+```
+
+## Restful Web Service
+
+- REST stands for Representational State Transfer. It's an architectural pattern for creating web services. A RESTful service is one that implements that pattern and follows the naming conventions of big to small
+
+- Example: dog/{id}/owner/{id}  (with this we can see that owner is a part of dog)
```

Comment:
The best practice would be something like: owners/{id}/dogs/{id}
From bigger to small (owners have dogs), and using plurals rather than singular .